e2e-tests/tests/committee/conftest.py

```diff
@@ -441,14 +442,21 @@ def _inner(mc_epoch):
                 probability_of_selection_in_single_epoch = (permissioned_seats / total_committee_seats) * (
                     PERMISSIONED_CANDIDATE_WEIGHT / permissioned_candidates_count
                 )
+                expected_attendance = permissioned_seats * (
+                    PERMISSIONED_CANDIDATE_WEIGHT / permissioned_candidates_count
+                )
+                guaranteed_seats = math.floor(expected_attendance)
             else:
                 probability_of_selection_in_single_epoch = (trustless_seats / total_committee_seats) * (
                     candidate.stake_delegation / total_stake
                 )
+                expected_attendance = trustless_seats * (candidate.stake_delegation / total_stake)
+                guaranteed_seats = math.floor(expected_attendance)
             expected_attendance_in_single_epoch = total_committee_seats * probability_of_selection_in_single_epoch
             total_expected_attendance = epochs_num * expected_attendance_in_single_epoch
             candidate.probability = probability_of_selection_in_single_epoch
```

Comment:
If you really need `.probability`, then I think it is `min(1, expected_attendance)`